### ISSUE TITLE
Use matrix Application Service API (matrix)

### ISSUE THREAD

**Issue by 42wim** [42wim/matterbridge#137]
Look into using matrix application service api for the matrix bridge.
(https://matrix.org/docs/guides/application_services.html)

**Comment by t3chguy** [comment]
I believe we have no go helper for Application Services yet, but gomatrix iirc includes the AS extension to the CS API so would just need a http server listening on specific routes to receive the pushed events from Synapse.

**Comment by xludx** [comment]
Could this be of any help: https://github.com/tulir/matrix-appservice-go

**Comment by benrob0329** [comment]
Perhaps using something like [matrix-appservice-webhooks](https://github.com/turt2live/matrix-appservice-webhooks) could be better.

**Comment by MayhemBill** [comment]
Has there been any update to this?

**Comment by 42wim** [comment]
@MayhemBill no, I think the number of users who would actually use this would be rather small, you'll need to setup your own homeserver for this. So it's not priority for me at the moment.

But if someone sends me a PR that adds support for this via eg https://github.com/tulir/mautrix-appservice-go , I will accept it gladly.

**Comment by TheHolyRoger** [comment]
Is using swagger not a better approach? I might look into this.

https://github.com/go-swagger/go-swagger
https://matrix.org/docs/api/client-server/

**Comment by t-my** [comment]
Application API requires hosting matrix homeserver and accessing api's directly. Remote access is forbidden for security reasons so this might be unfeasible.

**Comment by w3bb** [comment]
With Dendrite, a small homeserver entering beta, and various lightweight implementations supporting the appservice API now this might be more worth it. You can easily run a small bridging server on a 5 buck VPS.

**Comment by w3bb** [comment]
I might give this a shot

**Comment by hellcp** [comment]
This would be really useful for us
